shell/browser/api/electron_api_web_frame_main.cc

```diff
@@ -296,6 +296,12 @@ GURL WebFrameMain::URL() const {
   return render_frame_->GetLastCommittedURL();
 }
 
+std::string WebFrameMain::Origin() const {
+  if (!CheckRenderFrame())
+    return std::string();
```

Comment:
Does this make it possible for `frame.origin` to return an empty string, rather than a tuple or `null`?

Reply:
no, in this case an exception is thrown. (See CheckRenderFrame.)